Assets/Thirdweb/Examples/Scripts/Prefab_ConnectWallet.cs

```diff
@@ -0,0 +1,256 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using Thirdweb;
+using System;
+using TMPro;
+
+// Can be added to SDK
+public enum Chain
+{
+    Ethereum,
+    Goerli,
+    Polygon,
+    Mumbai,
+    Arbitrum,
+    ArbitrumGoerli,
+    Avalanche,
+    AvalancheFujiTestnet,
+    BinanceSmartChainMainnet,
+    BinanceSmartChainTestnet,
+    Fantom,
+    FantomTestnet,
+    Optimism,
+    OptimismGoerli,
+}
+
+// Can be added to SDK
+[Serializable]
+public enum Wallet
+{
+    MetaMask,
+    CoinbaseWallet,
+    WalletConnect,
+    Injected,
+    MagicAuth,
+}
+
+// Can be replaced with Serializeable Dictionary (asset)
+[Serializable]
+public struct WalletButton
+{
+    public Wallet wallet;
+    public GameObject walletButton;
+}
+
+public class Prefab_ConnectWallet : MonoBehaviour
+{
+    [Header("SETTINGS")]
+    public Chain chain;
+    public List<Wallet> supportedWallets;
+
+    [Header("UI - CONNECTING")]
+    public GameObject connectButton;
+    public Transform connectDropdown;
+    public List<WalletButton> walletButtons;
+
+    [Header("UI - CONNECTED")]
+    public GameObject connectedButton;
+    public Transform connectedDropdown;
+    public TMP_Text connectInfoText;
+    public TMP_Text walletAddressText;
+
+    bool dropdownExpand;
+    Coroutine dropdownRoutine;
+    string address;
+
+    ThirdwebSDK SDK;
+
+    // SDK Initialization
+
+    private void Awake()
+    {
+#if !UNITY_EDITOR
+        SDK = new ThirdwebSDK(chain.ToString().ToLower());
+#endif
+    }
+
+    // UI Initialization
+
+    private void Start()
+    {
+        address = null;
+
+        foreach (WalletButton wb in walletButtons)
+            wb.walletButton.SetActive(supportedWallets.Contains(wb.wallet));
+
+        connectDropdown.localScale = Vector3.zero;
+        connectDropdown.gameObject.SetActive(false);
+
+        connectedDropdown.localScale = Vector3.zero;
+        connectedDropdown.gameObject.SetActive(false);
+
+        connectButton.SetActive(true);
+        connectedButton.SetActive(false);
+
+        dropdownExpand = false;
+    }
+
+    // Connecting
+
+    public async void OnConnect(Wallet _wallet)
+    {
+        WalletProvider _provider = GetWalletProvider(_wallet);
+        int _chainID = GetChainID(chain);
+
+        try
+        {
+            address = await SDK.wallet.Connect(
+               new WalletConnection()
+               {
+                   provider = _provider,
+                   chainId = _chainID
+               });
+
+            connectInfoText.text = $"{_wallet} ({chain})";
+            walletAddressText.text = $"Connected As: {address}";
```

Comment:
can we shorten the address? it makes the button really large otherwise. We can have a util that shortens to `0x123...456`